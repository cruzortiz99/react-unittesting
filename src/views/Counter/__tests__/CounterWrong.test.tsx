import CounterWrong from "../WrongCounter";
import { render, fireEvent, screen, cleanup } from "@testing-library/react";

describe("CounterWrong component", () => {
  afterEach(() => cleanup());
  test("Should have 0 as default value when start render", async () => {
    const { container } = render(<CounterWrong />);
    const defaultValueExpected = await screen.findByText(/0/);
    expect(defaultValueExpected).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  test("Should add 1 to the count when add button is clicked", async () => {
    const { container } = render(<CounterWrong />);
    const button = await screen.findByText("Add");
    fireEvent.click(button);
    const count = await screen.findByText(/1/);
    expect(count).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  test("Should show the correct sum when the add button is clicked several times", async () => {
    await [3,15,1000].map((times) => async () => {
      render(<CounterWrong />);
      const button = await screen.findByText("Add");
      const total = times;
      for(let i = 0; i < times; i++) {
        fireEvent.click(button)
      }
      const count = await screen.findByText(RegExp(`${total}`));
      expect(count).toBeTruthy();
      cleanup()
    }).reduce((prevTest, currentTest) => prevTest.then(() => currentTest()), Promise.resolve())
  });
  test("Should reduce 1 to the count when minus button is clicked", async () => {
    const { container } = render(<CounterWrong />);
    const button = await screen.findByText("Minus");
    fireEvent.click(button);
    const count = await screen.findByText(/-1/);
    expect(count).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  test("Should show the correct count when the minus button is clicked several times", async () => {
    await [3,15,1000].map((times) => async () => {
      render(<CounterWrong />);
      const button = await screen.findByText("Minus");
      const total = 0 - times;
      for(let i = 0; i < times; i++) {
        fireEvent.click(button)
      }
      const count = await screen.findByText(RegExp(`${total}`));
      expect(count).toBeTruthy();
      cleanup()
    }).reduce((prevTest, currentTest) => prevTest.then(() => currentTest()), Promise.resolve())
  });
});
