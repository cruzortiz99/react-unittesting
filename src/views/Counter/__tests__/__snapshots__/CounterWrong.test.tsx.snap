// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterWrong component Should add 1 to the count when add button is clicked 1`] = `
<div>
  <div>
    <h3>
      Wrong Counter
    </h3>
    <p>
      Count: 
      1
    </p>
    <button>
      Add
    </button>
    <button>
      Minus
    </button>
  </div>
</div>
`;

exports[`CounterWrong component Should have 0 as default value when start render 1`] = `
<div>
  <div>
    <h3>
      Wrong Counter
    </h3>
    <p>
      Count: 
      0
    </p>
    <button>
      Add
    </button>
    <button>
      Minus
    </button>
  </div>
</div>
`;

exports[`CounterWrong component Should reduce 1 to the count when minus button is clicked 1`] = `
<div>
  <div>
    <h3>
      Wrong Counter
    </h3>
    <p>
      Count: 
      -1
    </p>
    <button>
      Add
    </button>
    <button>
      Minus
    </button>
  </div>
</div>
`;
